daffodil-runtime1/src/main/scala/org/apache/daffodil/infoset/InfosetWalker.scala

```diff
@@ -408,14 +413,23 @@ class InfosetWalker private (
     containerIndexStack.setTop(top + 1)
   }
 
+  @inline
+  private def doOutputter(outputterFunc: => Boolean, desc: String, context: ThrowsSDE): Unit = {
+    Try(outputterFunc) match {
+      case Success(true) => // success
+      case Success(false) => context.SDE("Failed to %s", desc)
+      case Failure(e) => context.SDE("Failed to %s: %s", desc, e.toString)
```

Comment:
Did you consider whether this should just be encapsulating the error as a XMLTextInfosetOutputterFailure(e) and throwing that to the caller inside Daffodil?

I'm not sure that is better, just wondering.

Reply:
That might make more sense, that makes it more clear that the problem is in the infoset inputter which is user provided, and not necessarily a problem with Daffodil. Though, most people are going to use the ones we ship with Daffodil, so it might be unexpected for users to have to catch an exception instead of just get a normal SDE. I don't feel strongly either way.

Reply:
I think throwing the exception might cause problems--I don't think parse ever is supposed to throw, we just always convert exceptions to some form of diagnostic or parse/unaprse error. But I can update the SDE to include the reason.